ts/quickjs.ts

```diff
@@ -102,6 +103,22 @@ export class Lifetime<T, Owner = never> {
     return this._owner
   }
 
+  /**
+   * Create a new handle pointing to the same [[value]].
+   */
+  dup() {
+    this.assertAlive()
+    if (!this.copier) {
+      throw new Error('Non-copiable lifetime')
```

Comment:
I don't know if that's exactly what you mean, but I spent a bit of time playing with the concept of JSValuePointer and JSValueConstPointer in order to make them compatible (so that non-const could be downcasted to const transparently), and it was really painful and didn't work out in the end 🤔

I think for now I'd prefer to implement the `canDup`, and iterate later if needed.

Reply:
What I’m asking here is can we know, statically, which Lifetime can be duped or not? I don’t see a reason why we can’t know statically. All the lifetimes a user will access will be dupable, right? All the QuickJS handles are, I think - so we can make the dup function a required argument for some kinds of Lifetime, and then there’s no need for a canDup(). 

The point of this is to make it clear when a method of Lifetime will throw an exception. Many uses of Lifetime can throw an exception, but the user can avoid all of the existing cases by checking if `lifetime.alive` before using the lifetime.

canDispose() isn’t necessary because dispose() won’t throw an exception if a lifetime doesn’t have a disposer.

Reply:
Maybe all this is unnecessary because in practice dup() will never throw for the user.

Reply:
There are a few specific instances where a lifetime is non-copiable: the ones returned by `createVm` and `toPointerArray`. Do you still prefer to revert `canDup`?

Reply:
Yes, because those lifetimes are all private and internal to the library.